src/Arcus.Security.Providers.KeyPerFile/KeyPerFileSecretProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using System;
+using System.Threading.Tasks;
+
+namespace Arcus.Security.Providers.KeyPerFile
+{
+    public class KeyPerFileSecretProvider : ISecretProvider
+    {
+        private readonly KeyPerFileConfigurationProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KeyPerFileSecretProvider"/> class.
+        /// </summary>
+        public KeyPerFileSecretProvider(KeyPerFileConfigurationSource configurationSource)
+        {
+            Guard.NotNull(configurationSource, nameof(configurationSource));
+            _provider = new KeyPerFileConfigurationProvider(configurationSource);
+            _provider.Load();
```

Comment:
Reason why this is called inside the constructor and not in the extension? Because it allows you to make sure that the configuration provider was loaded?

Reply:
It makes sure that if I instantiate an instance, the secrets are loaded as well.
It might make sense to move it outside the ctor, but then users will to call the Load method themselves.   I can change that if you want

Reply:
This is ok for me! But does this mean it will not reflect changes to the file later on?

Reply:
The UserSecrets is different in that way. It's begin called there before the ctor. Seems like they both should have the same setup but not sure which one.
Not sure also if it should be the responsibility of the secret provider that the configuration was loaded. 🤔 If this could be checked or so... Also, with 'reloading' this should maybe looked further into.

Reply:
Well, the ´Load´ method is part of the KeyPerFileConfigurationProvider whereas the parameter of the ctor is a configuration source.
I could change that we're injecting a configurationprovider instead of a configuration source though.

Reply:
Would we then have to still load it or just act as we call it?

Reply:
What do you mean ?  Call load when the first secret is retrieved ?

Reply:
Yes or will it work automatically with ConfigurationProvider? Or does that depend on the IConfiguration of the user? Sorry I'm lost here I think.

Reply:
Calling `Load` on the provider loads all files that are present in the configured directory, and creates 'secrets' from them.  So, if you create the secret provider, all secrets will be loaded only once.  
In contrast of the `KeyVaultSecretProvider`which requests the secret every time from KeyVault itself (when no caching is configured), this provider retrieves the secrets only once.

This can be changed offcourse, but  then we should not rely on the `KeyPerFile` configuration provider which is now used internally, but implement this functionality ourselves.  In that case, we would need to load the correct file, read and return its contents on every call to 'GetSecretAsync'.

Reply:
@tomkerkhove what is your view on this ?

Reply:
> Question is, is this necessary ? When using docker and docker secrets, the secrets are mounted as files inside the container. When the secret changes, I guess that this also means that the container needs to be restarted ?

That is not the case actually, often hot-swapping of configs are done without restarting the container.

Maybe odd question, but what if we call Load now and then to refresh things?